builtin/provisioners/chef/resource_provisioner.go

```diff
@@ -703,7 +703,16 @@ func (p *provisioner) runCommand(o terraform.UIOutput, comm communicator.Communi
 	}
 
 	if err := cmd.Wait(); err != nil {
-		return err
+		// Allow RFC062 Exit Codes
+		// https://github.com/chef/chef-rfc/blob/master/rfc062-exit-status.md
+		switch err.ExitStatus {
+		case 35:
+			o.Output("Reboot has been scheduled in the run state")
+		case 37:
+			o.Output("Reboot needs to be completed")
+		default:
+			return err
+		}
```

Comment:
Guess it needs something like this:

```suggestion
		exitErr, ok := err.(remote.ExitError)
		if !ok {
			return err
		}
		
		// Allow RFC062 Exit Codes
		// https://github.com/chef/chef-rfc/blob/master/rfc062-exit-status.md
		switch exitErr.ExitStatus {
		case 35:
			o.Output("Reboot has been scheduled in the run state")
		case 37:
			o.Output("Reboot needs to be completed")
		default:
			return exitErr
		}
```

Reply:
Minor fix needed (and var name change to keep in line with the Puppet provisioner):

```suggestion
		// Allow RFC062 Exit Codes
		// https://github.com/chef/chef-rfc/blob/master/rfc062-exit-status.md
		errStruct, _ := err.(*remote.ExitError)
		switch errStruct.ExitStatus {
		case 35:
			o.Output("Reboot has been scheduled in the run state")
		case 37:
			o.Output("Reboot needs to be completed")
		case 213:
			o.Output("Chef has exited during a client upgrade")
		default:
			return err
		}
```

I thought to include 213 as that's a controlled termination of the chef-client process and shouldn't be considered a failure.